src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -295,6 +297,9 @@ private ICitizenData spawnCitizenOnPosition(
         entity.setPosition(spawnPoint.getX() + HALF_BLOCK, spawnPoint.getY() + SLIGHTLY_UP, spawnPoint.getZ() + HALF_BLOCK);
         world.addEntity(entity);
 
+        colony.getEventManager().addEventDescription(citizenData.isChild() ? new CitizenBornEvent(spawnPoint, citizenData.getName()) :
+              new CitizenSpawnedEvent(spawnPoint, citizenData.getName()));
```

Comment:
this is also used for entity respawning, which would trigger the event again

Reply:
"entity respawning" as in loading the world again, or recalling, or...?

Reply:
loading the world no, but if its missing or gets recalled to respawn etc, minecraft sometimes looses the entities and we need to respawn them

Reply:
if the citizen goes missing, does `citizenData.getEntity()` return null, or an invalid entity?

Reply:
could be either

Reply:
BuildingHome has the child spawn, and the RecruitmentInteraction has the visitors

Reply:
so the only solution is to move the event handling to the classes requesting the citizen spawn instead of the spawn handler?

Reply:
well spawning is only the entity, not the citizen itself. Entities are exchangeable/get lost, citizenData is the actualy citizen

Reply:
wait, does that mean i can handle that in the CitizenData constructor?

Reply:
no thats called upon loading too :D

Reply:
once the crash is fixed i will look more into this.

Reply:
seems like the recent citizen loading changes cause world loading to sometimes cause a citizen spawned event, but not always.

Reply:
I hope this should be completely done now.
I tested it some time and it seems to fully work.